src/Contract/Frequency.php

```diff
@@ -32,4 +33,27 @@ public function getNextDue(DateTimeInterface $dateTime): DateTimeInterface;
      * @return bool
      */
     public function isExpired(DateTimeInterface $dateTime): bool;
+
+    /**
+     * Get the start time of this frequency
+     *
+     * @return ?DateTimeInterface
+     */
+    public function getStart(): ?DateTimeInterface;
+
+    /**
+     * Get the end time of this frequency
+     *
+     * @return ?DateTimeInterface
+     */
+    public function getEnd(): ?DateTimeInterface;
+
+    /**
+     * Parse the given json object and derive a {@see Frequency} instance from it
+     *
+     * @param string $data
```

Comment:
I would use `$json` instead of `$data` since you get the data back from the JSON representation.